File: week12/week12-3.cpp
///week12-3.cpp 列出質數
#include <stdio.h>
int isPrime(int n)
{///質數不能整除
    for(int i=2;i<n;i++){
        if(n%i==0)return 0;///失敗
    }
    return 1;///成功!
}
int main()
{
    printf("請輸入1個數:");
    int n;
    scanf("%d",&n);
    int ans = 0;
    for(int i=2;i<=n;i++){
        if(isPrime(i)){
            printf("%d ",i);
            ans ++;
        }
    }
    printf("總共有 %d 個質數\n",ans);
}///228000有20260個質數

File: week12/week12-3.py
#week12-3.py
table = [True]*240000
ans = 0
for i in range(2, 240000):
  if table[i]==True:
    ans += 1
    for k in range(i*i, 240000, i): table[k] = False #國王繼位後,把他的兄弟殺光光

print('找到',ans,'個質數')